modules/apps/headless-apio/media-object/media-object-apio-impl/src/main/java/com/liferay/media/object/apio/internal/architect/resource/MediaObjectNestedCollectionResource.java

```diff
@@ -132,6 +139,91 @@ public String getName() {
 		).build();
 	}
 
+	private List<AdaptiveMedia<AMImageProcessor>> _getAdaptiveMedia(
+		FileEntry fileEntry) {
+
+		String mimeType = fileEntry.getMimeType();
+
+		if (_amImageMimeTypeProvider.isMimeTypeSupported(mimeType)) {
+			return Try.fromFallible(
+				() -> _amImageFinder.getAdaptiveMediaStream(
+					amImageQueryBuilder -> amImageQueryBuilder.forFileEntry(
+						fileEntry
+					).withConfigurationStatus(
+						AMImageQueryBuilder.ConfigurationStatus.ANY
+					).done()
+				).collect(
+					Collectors.toList()
+				)
+			).orElse(
+				null
+			);
+		}
+
+		return null;
+	}
+
+	private String _getAdaptiveMediaContentUrl(AdaptiveMedia adaptiveMedia) {
+		URI uri = adaptiveMedia.getURI();
+
+		return String.valueOf(uri);
+	}
+
+	private Integer _getAdaptiveMediaHeight(
+		AdaptiveMedia<AMImageProcessor> adaptiveMedia) {
+
+		Optional<Integer> height = adaptiveMedia.getValueOptional(
+			AMImageAttribute.AM_IMAGE_ATTRIBUTE_HEIGHT);
+
+		return height.orElse(0);
+	}
+
+	private String _getAdaptiveMediaName(
+		AdaptiveMedia<AMImageProcessor> adaptiveMedia) {
+
+		Optional<String> name = adaptiveMedia.getValueOptional(
+			AMAttribute.getConfigurationUuidAMAttribute());
+
+		return name.orElse(null);
+	}
+
+	private NestedRepresentor<AdaptiveMedia<AMImageProcessor>>
+		_getAdaptiveMediaNestedRepresentor(
+			Builder<AdaptiveMedia<AMImageProcessor>> builder) {
+
+		return builder.types(
+			"ImageObject", "MediaObject"
+		).addNumber(
+			"width", this::_getAdaptiveMediaWidth
+		).addNumber(
+			"height", this::_getAdaptiveMediaHeight
+		).addString(
+			"resolutionName", this::_getAdaptiveMediaName
+		).addNumber(
+			"sizeInBytes", this::_getAdaptiveMediaSize
+		).addRelativeURL(
+			"contentUrl", this::_getAdaptiveMediaContentUrl
+		).build();
+	}
+
+	private Number _getAdaptiveMediaSize(
```

Comment:
although it would be longer, I rather prefer a _getAdaptiveMediaValue(adaptiveMedia, AMAttribute.getContentLengthAMAttribute()) method instead of several methods that only extract values

Reply:
👍